pkg/util/util_linux.go

```diff
@@ -44,8 +45,11 @@ func CleanupSocketFile(socketPath string) error {
 
 func DoesSocketExist(socketPath string) (bool, error) {
 	fi, err := os.Stat(socketPath)
-	if err == nil && (fi.Mode()&os.ModeSocket) != 0 {
-		return true, nil
+	if err == nil {
+		if isSocket := (fi.Mode()&os.ModeSocket != 0); isSocket {
+			return true, nil
+		}
+		klog.Warningf("file exists in socketPath %s but it's not a socket.: %+v", socketPath, fi)
```

Comment:
Does this fix https://github.com/kubernetes-csi/node-driver-registrar/issues/104? It seems like this will return `false` and `nil` for the error if it is not a socket.

Reply:
Yeah good point... I initially return error but then wonder if it will break things so I change it to warning. I think an Error here seems okay to me? It's just we will have to change the CleanUpSocket function as well.

Reply:
I vaguely recall the socket checking logic differs between linux and Windows. It might be worth double-checking.

Reply:
There is already a TestSocketRegularFile which I modified a little bit that test this behavior.
The limitation is that we will have different behaviors on linux and windows... Because according https://github.com/golang/go/issues/33357, windows socket file are not returning the right mode somehow... So the test will behavior a little bit different based on the two platforms.